src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -309,7 +309,9 @@ private async Task<SyncTriggersPayload> GetSyncTriggersPayload()
             PrepareSyncTriggers();
 
             var hostOptions = _applicationHostOptions.CurrentValue.ToHostOptions();
-            var functionsMetadata = _functionMetadataManager.GetFunctionMetadata().Where(m => !m.IsProxy());
+            var functionsMetadata = _functionMetadataManager.GetFunctionMetadata().Where(m => !m.IsProxy())
+                .GroupBy(m => m.Name, StringComparer.OrdinalIgnoreCase)
+                .Select(g => g.First());
```

Comment:
There's a second GetFunctionMetadata call in this method that is also listing functions (with a slightly different query) and turning them into an array. This isn't causing problems currently because apparently we don't have any code doing a ToDictionary on the values like we do in the other case.

Initially I was going to suggest we de-dupe there as well, but perhaps we don't need to. The "functions" section is supposed to represent all the listable functions, similar to our other "source of truth" arguments.

Reply:
`listableFunctions` is only referenced twice in this file (when value is assigned, when added as an array to full syncTriggers payload as you mentioned). The one I de-deduped is used for triggers and secrets (which are more important to de-dupe). 

I think having the "source of truth" in this case is fine since it shouldn't block sync triggers updates like this bug did. Using the listable functions (probably used for display in portal?) could help customers figure out they have a problem more quickly.